src/osdc/Objecter.cc

```diff
@@ -2933,10 +2923,21 @@ void Objecter::_session_op_assign(OSDSession *to, Op *op)
   op->session = to;
   to->ops[op->tid] = op;
 
+  if(op->on_noosd_timeout && (! to->is_homeless())){
+    timer.cancel_event(op->on_noosd_timeout);
+    op->on_noosd_timeout = 0;
+    ldout(cct, 0) << __func__ << " session timeout cancelled" << dendl;
+  }
+
   if (to->is_homeless()) {
     num_homeless_ops++;
+    if (cct->_conf->objecter_homeless_timeout > 0) {
+      op->on_noosd_timeout = timer.add_event(ceph::make_timespan(cct->_conf->objecter_homeless_timeout), [this, tid = op->tid]() {
+        _op_cancel(tid, -ETIMEDOUT);
+        ldout(cct, 0) << __func__ << " Op cancellation due to session timeout" << dendl;
```

Comment:
i don't think log level 0 is appropriate for these messages. these aren't fatal errors that require admin intervention, and admins can't disable log level 0 if it's spamming their client logs. it looks like 5 or 10 would be more consistent with other messages in Objecter

Reply:
Changed it to 5